lib/openshift/cluster_logging.rb

```diff
@@ -69,11 +93,22 @@ class ClusterLogging < ClusterResource
     end
 
     # higher level methods
+    def rsyslog_ready?(user: nil, cached: true, quiet: false)
+      rsyslog_nodes = rsyslog_status_raw(user: user, cached: cached, quiet: quiet)['Nodes'].keys.sort
+      rsyslog_nodes == rsyslog_ready_pods && rsyslog_failed_pods.count == 0 && rsyslog_notready_pods.count == 0
+    end
+
     def fluentd_ready?(user: nil, cached: true, quiet: false)
       fluentd_nodes = fluentd_status_raw(user: user, cached: cached, quiet: quiet)['nodes'].keys.sort
       fluentd_nodes == fluentd_ready_pods && fluentd_failed_pods.count == 0 && fluentd_notready_pods.count == 0
     end
 
+    def wait_until_rsyslog_is_ready(user: nil, quiet: false, timeout: 5*60)
+      wait_for(timeout) {
+        rsyslog_ready?(user: user, cached: false, quiet: quiet)
+      }
```

Comment:
should this fail when timeout is reached. Presently it will just exit.

Reply:
I'll add steps to raise error in logging.rb when timeout is reached.

Reply:
@akostadinov I added a statement to raise error when timeout is reached in logging.rb. And the message would be:
 ```  Failed to wait for rsyslog to be ready (RuntimeError)
      /home/qitang/github/verification-tests/features/step_definitions/logging.rb:135:in `/^I wait for clusterlogging(?: named "(.+)")? with "(.+?)" log collector to be functional in the(?: "(.+?)")? project$/'
      /home/qitang/github/verification-tests/features/step_definitions/logging.rb:255:in `/^I create clusterlogging instance with:$/'
      features/logging/test.feature:28:in `Given I create clusterlogging instance with:'
```

Reply:
sorry, can't see these in here

Reply:
Please look at: https://github.com/openshift/verification-tests/pull/340/files#diff-0f0202f69f619cc471255d933a81861cR139-R140 and https://github.com/openshift/verification-tests/pull/340/files#diff-0f0202f69f619cc471255d933a81861cR145-R146

Reply:
What I request is to make this method fail in case rsyslog never does not become ready. It is needed and it is less reliable to ask all users of the method to do a check after calling it. When you do it here, all users of the method will fail when waiting is not successful.

Reply:
I mean somehting like:
```
    def wait_until_rsyslog_is_ready(user: nil, quiet: false, timeout: 5*60)
      success = wait_for(timeout) {
        rsyslog_ready?(user: user, cached: false, quiet: quiet)
      }
      unless success
        raise "rsyslog did not become ready within #{timeout} seconds"
    end
```